### PR TITLE
#171424328 Refactor multiCity trip request

### PR DESCRIPTION
**Pull request by AbdoulNuru**
### What does this PR do?

- Change the way a multi-city trip request is made.
- validate the request.

### How should this be manually tested?

- clone this repository by running `git clone <repo link>`
- run this command `git checkout <branch name>` in your terminal to navigate to this branch `ft-multi-city-request-170947442`
- create a *.env* file and reference *.env.example* to add all the variable.
- run `npm run migrateDb` to run the migrations
- run `npm run dev` to start the development server
- using postman test this route `/api/v1/trips/multi-city`, with a post method

### Implementation plan

[MultiCity trip request](https://docs.google.com/document/d/15IqXJj_YWhyYqcG1m3eSmnU3kIxjiTEaQkHaoq9Llfw/edit)

### Stories associated with PR

[#171424328](https://www.pivotaltracker.com/story/show/171424328)

### Screenshots if any
Screen Shot 2020-02-24 at 23 13 36

### CODE REVIEW

**Review by blaise82 — APPROVED** [review]

File: locales/en.js [hunk 1]
```diff
@@ -14,32 +14,52 @@
...
+	"mStopsDepDate": "Please provide valid date as a departure date and date format must be YYYY-MM-DD",
+	"mRetDate": "Please provide valid date as a return date and date format must be YYYY-MM-DD",
+	"sCountry": "stopCountry should not be empty. Please use country letter representations example [KN, Rw, USA, UG, ...]",
+	"mAccommodation": "accommodation should not be empty.Please if you need accommodation enter [yes] if no enter [no]",
+	"mArrivalDate": "Enter a valid arrival date and the date format must be YYYY-MM-DD",
+	"mReason": "Please provide a reason for stoping in this city",
+	"notSuperAdmin": "You have to be loggen in as a super administrator to assign roles to users",
+	"userDon'tExist": "The User you are assigning the role to don't exist",
+	"roleUpgrade": "Your role was upgraded to"
+}
```
Comment (hakoemmy) [inline locales/en.js]:
```suggestion
}

```

File: locales/fr.js [hunk 2]
```diff
@@ -1,44 +1,61 @@
...
+	"mStopsDepDate": "Veuillez fournir une date valide comme date de départ et le format de la date doit être AAAA-MM-JJ",
+	"sCountry": "stopCountry ne doit pas être vide. Veuillez utiliser l'exemple de représentation des lettres du pays [KN, Rw, USA, UG, ...]",
+	"mAccommodation": "le logement ne doit pas être vide. S'il vous plaît si vous avez besoin d'un logement, entrez [oui] si non entrez [non]",
+	"mArrivalDate": "Entrez une date d'arrivée valide et le format de la date doit être AAAA-MM-JJ",
+	"mReason": "Veuillez indiquer la raison de votre arrêt dans cette ville",
+	"Please provide a reason for stoping in this city": "Please provide a reason for stoping in this city",
+	"notSuperAdmin": "Vous devez être connecté en tant que super administrateur pour attribuer des rôles aux utilisateurs",
+	"userDon'tExist": "L'utilisateur auquel vous attribuez le rôle n'existe pas",
+	"roleUpgrade": "Votre rôle a été mis à niveau vers"
+}
```
Comment (hakoemmy) [inline locales/fr.js]:
```suggestion
}

```

File: src/controllers/userController.js [hunk 3]
```diff
@@ -203,6 +183,12 @@ class userController {
           error: i18n.__('IncorrectEmailPassword'),
         });
       }
+      if (!registered.isUpdated) {
+        return res.status(403).json({
+          status: 403,
+          error: 'Please update your profile information to continue'
```
Comment (hakoemmy) [inline src/controllers/userController.js:189]:
```suggestion
       error:  setLanguage(preferedLang).__('Please_update_your_profile_information_to _continue')
```
@AbdoulNuru Would you please make translation for that response.

You'll have to use such ```setLanguage``` method when I merge. But at least set the locales

File: src/routes/api/trips.routes.js [hunk 4]
```diff
@@ -78,37 +78,44 @@ router.post('/request', [checkUser, dateValidator, tripRequestValidator], tripsC
...
- *                 departureDate:
- *                   type: string
- *                 reason:
- *                   type: string
+ *             type: object
+ *             properties:
+ *               type:
+ *                 type: string
+ *               fromCountry:
+ *                 type: string
```
Comment (hakoemmy) [inline src/routes/api/trips.routes.js:93]:
```suggestion
 *                 type: string
                    example: Rwanda
```
It's better to show examples for every property while documenting. It'll simplify the work of someone who will present on the demo

File: src/controllers/userController.js (hunk 3 above)
Comment (AbdoulNuru) [reply src/controllers/userController.js:189]: @hakoemmy I can't use a method that is not there, what is there works I will refactor once you merge your PR, but now I can not use a method that is not available

File: src/routes/api/trips.routes.js (hunk 4 above)
Comment (AbdoulNuru) [reply src/routes/api/trips.routes.js:93]: @hakoemmy I have validated every input.

File: locales/fr.js (hunk 2 above)
Comment (AbdoulNuru) [reply locales/fr.js]: Thank you, I added the new line

Comment (hakoemmy) [reply locales/fr.js]: @AbdoulNuru Good to know

File: src/controllers/userController.js (hunk 3 above)
Comment (hakoemmy) [reply src/controllers/userController.js:189]: @AbdoulNuru Sure thing

File: src/routes/api/trips.routes.js (hunk 4 above)
Comment (hakoemmy) [reply src/routes/api/trips.routes.js:93]: @AbdoulNuru I understand, it was my suggestion. But still, it's OK

**Review by RusimbiPatrick — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

**Review by hakoemmy — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**